### PR TITLE
[Audio] fix setAudioModeAsync to auto-fill default values

### LINKED ISSUE #5456: [iOS][Audio] Playing sound in background Problem
### Environment

  Expo CLI 3.0.2 environment info:
    System:
      OS: macOS 10.14.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.16.0 - /usr/local/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Xcode: 10.3/10G8 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^33.0.0 => 33.0.5 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.0 => 3.11.0 
    npmGlobalPackages:
      expo-cli: 3.0.2

>

### Steps to Reproduce

1- Open app, play sound
2- Lock Phone
3- Stop or pause sound with Music controls buttons
4- Try to play sound with Music controls buttons

Error: 
- code: E_AV_PLAY 
- domain: UMModulesErrorDomain 
- NSLocalizedDescription: This experience is currently in the background, so the audio session could not be activated

I found the related file and comment that line works for me. Now I can play and stop sounds on background on IOS.

File; [https://github.com/expo/expo/blob/master/packages/expo-av/ios/EXAV/EXAV.m#L272](https://github.com/expo/expo/blob/master/packages/expo-av/ios/EXAV/EXAV.m#L272) 

```
  if (_isBackgrounded && ![_kernelAudioSessionManagerDelegate isActiveForModule:self]) {
    //return UMErrorWithMessage(@"This experience is currently in the background, so the audio session could not be activated.");
  }
```

### Sample
```
import { Audio } from "expo-av"
import MusicControl from "react-native-music-control"


Audio.setAudioModeAsync({
	allowsRecordingIOS: false,
	staysActiveInBackground: true,
	interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
	playsInSilentModeIOS: true,
	shouldDuckAndroid: false,
	interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX,
	playThroughEarpieceAndroid: true
})

MusicControl.resetNowPlaying()
MusicControl.enableControl("play", true)
MusicControl.enableControl("pause", true)
MusicControl.enableControl("stop", true)
MusicControl.enableControl("nextTrack", false)
MusicControl.enableControl("previousTrack", false)
MusicControl.enableControl("changePlaybackPosition", false)
MusicControl.enableControl("setRating", false)
MusicControl.enableControl("volume", false)
MusicControl.enableControl("remoteVolume", false)
MusicControl.enableControl("enableLanguageOption", false)
MusicControl.enableControl("disableLanguageOption", false)
MusicControl.enableControl("closeNotification", true, { when: "always" })

MusicControl.enableBackgroundMode(true)
MusicControl.handleAudioInterruptions(true)
MusicControl.on("play", () => {
	playSound()
})
MusicControl.on("pause", () => {
	stopSound()
})
MusicControl.on("stop", () => {
	stopSound()
})
MusicControl.on("closeNotification", () => {
	stopSound()
})
```

### PR DESCRIPTION
**Pull request by cruzach**
# Why

the docs state that there are default values for these params -> https://docs.expo.io/versions/latest/sdk/audio/#audiosetaudiomodeasyncmode
but if every single one isn't provided (including both Android- and iOS-specific keys), an error is thrown

closes #5456 

# How

check object of params provided by user against default AudioMode object, if it doesn't have a key then we add that key with the associated default value

# Test Plan

tested locally

### CODE REVIEW

**Comment by cruzach** [comment]
`setAudioMode()` changes are now merged with the previously set AudioMode (if there is one)
Haven't done all the testing I want to on this yet, but feel free to review and give me your thoughts

File: packages/expo-av/src/Audio.ts [hunk 1]
```diff
@@ -28,25 +28,38 @@ const _isValueValid = (value: any, validValues: any[]): boolean => {
...
+  allowsRecordingIOS: false,
+  interruptionModeIOS: INTERRUPTION_MODE_IOS_MIX_WITH_OTHERS,
+  playsInSilentModeIOS: false,
+  staysActiveInBackground: false,
+  interruptionModeAndroid: INTERRUPTION_MODE_ANDROID_DUCK_OTHERS,
+  shouldDuckAndroid: true,
+  playThroughEarpieceAndroid: true,
+};
+
+let currentAudioMode: any = null;
```
Comment (sjchmiela) [inline packages/expo-av/src/Audio.ts]:
```suggestion
let currentAudioMode: AudioMode | null = null;
```

**Review by sjchmiela — APPROVED** [review]
I don't see where this approach could fail, so (apart from that one comment) 👍 🙂

File: packages/expo-av/src/Audio.ts (hunk 1 above)
Comment (cruzach) [reply packages/expo-av/src/Audio.ts]: Awesome! Thanks for the help and suggestions on this 😄

**Pull request merged**